handlers/dimensions.go

```diff
@@ -376,16 +374,33 @@ func sortedTime(opts dataset.Options) dataset.Options {
 			newList = append(newList, o.option)
 		}
 	}
-
 	//check lists are the same length (so contain the same data) and only
 	//return the sorted list if it's complete
 	if len(newList) == len(opts.Items) {
 		opts.Items = newList
 	}
-
 	return opts
 }
 
+func splitCode(id string) (string, string, error) {
+	code := strings.Split(id, "-")
+	if len(code) == 1 {
+		return "", "", errors.New("code cannot be split")
+	}
+
+	if len(code) < 3 {
+		return "", "", errors.New("code does not match expected format")
+	}
+
+	month := code[1]
```

Comment:
given (below) that the `year := code[len(code)-1]` i.e. the 'last' item in `code`,
shouldn't `month` therefore use the 'last but one' item in `code` (the one before year?) - to be more consistent? i.e.
`month := code[len(code)-2]`